bigquery/src/main/java/org/apache/iceberg/gcp/bigquery/BigQueryMetastoreCatalog.java

```diff
@@ -0,0 +1,400 @@
+      if (testingEnabled) {
+        client = new FakeBigQueryMetastoreClient(options);
+
+      } else {
+        client = new BigQueryMetastoreClientImpl(options);
+      }
+    } catch (IOException e) {
+      throw new ServiceFailureException(e, "Creating BigQuery client failed");
+    } catch (GeneralSecurityException e) {
+      throw new ValidationException(e, "Creating BigQuery client failed due to a security issue");
+    }
+    initialize(inputName, properties, projectId, location, client);
+  }
+
+  @VisibleForTesting
+  void initialize(
+      String inputName,
+      Map<String, String> properties,
+      String initialProjectId,
+      String initialLocation,
+      BigQueryMetastoreClient bigQueryMetaStoreClient) {
+    Preconditions.checkArgument(bigQueryMetaStoreClient != null, "BigQuery client can not be null");
+    this.catalogName = inputName;
+    this.catalogProperties = ImmutableMap.copyOf(properties);
+    this.projectId = initialProjectId;
+    this.location = initialLocation;
+    this.client = bigQueryMetaStoreClient;
+
+    if (this.conf == null) {
+      LOG.warn("No configuration was set, using the default environment Configuration");
+      this.conf = new Configuration();
+    }
+
+    LOG.info("Using BigQuery Metastore Iceberg Catalog: {}", inputName);
+
+    if (properties.containsKey(CatalogProperties.WAREHOUSE_LOCATION)) {
+      // Iceberg always removes trailing slash to avoid paths like "<folder>//data" in file systems
+      // like s3.
+      this.conf.set(
+          HIVE_METASTORE_WAREHOUSE_DIR,
+          LocationUtil.stripTrailingSlash(properties.get(CatalogProperties.WAREHOUSE_LOCATION)));
+    }
+
+    String fileIoImpl =
+        properties.getOrDefault(
+            CatalogProperties.FILE_IO_IMPL, "org.apache.iceberg.hadoop.HadoopFileIO");
+    this.fileIO = CatalogUtil.loadFileIO(fileIoImpl, properties, conf);
+
+    this.filterUnsupportedTables =
+        Boolean.parseBoolean(properties.getOrDefault(FILTER_UNSUPPORTED_TABLES, "false"));
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier identifier) {
+    return new BigQueryTableOperations(
+        client,
+        fileIO,
+        projectId,
+        // Sometimes extensions have the namespace contain the table name too, so we are forced to
+        // allow invalid namespace and just take the first part here like other catalog
+        // implementations do.
+        identifier.namespace().level(0),
+        identifier.name(),
+        conf);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier identifier) {
+    String locationUri = null;
+    DatasetReference datasetReference = toDatasetReference(identifier.namespace());
+    Dataset dataset = client.getDataset(datasetReference);
+    if (dataset != null && dataset.getExternalCatalogDatasetOptions() != null) {
+      locationUri = dataset.getExternalCatalogDatasetOptions().getDefaultStorageLocationUri();
+    }
+    return String.format(
```

Comment:
nit: please add a newline after a closing }. See also https://iceberg.apache.org/contribute/#block-spacing

Reply:
I updated where I see this issue